main.tf

```diff
@@ -74,3 +74,40 @@ resource "google_monitoring_uptime_check_config" "https_uptime" {
     create_before_destroy = true
   }
 }
+
+# ----------------------------
+# SSL expiration alert policy
+# ----------------------------
+resource "google_monitoring_alert_policy" "ssl_expiring_days" {
+  for_each = toset([for days in var.ssl_alert_threshold_days : tostring(days)])
+
+  display_name = "SSL certificate expiring soon (${each.value} days)"
+  combiner     = "OR"
+  conditions {
+    condition_threshold {
+      filter          = "metric.type=\"monitoring.googleapis.com/uptime_check/time_until_ssl_cert_expires\" AND resource.type=\"uptime_url\""
+      comparison      = "COMPARISON_LT"
+      threshold_value = each.value
+      duration        = "600s"
+      trigger {
+        count = 1
+      }
+      aggregations {
+        alignment_period     = "1200s"
+        per_series_aligner   = "ALIGN_NEXT_OLDER"
+        cross_series_reducer = "REDUCE_MEAN"
+        group_by_fields = [
+          "resource.label.*"
+        ]
+      }
+    }
+    display_name = "SSL certificate expiring soon (${each.value} days)"
+  }
+
+  user_labels = {
+    version = "1"
+    uptime  = "ssl_cert_expiration"
+  }
+
+  notification_channels = var.alert_notification_channels
```

Comment:
is this a known variable ?

Reply:
Yes, it this the same variable used for the `failure_alert` for the http checks ([used here](https://github.com/sparkfabrik/terraform-sparkfabrik-gcp-http-monitoring/blob/main/main.tf#L34)).